src/MahApps.Metro/MahApps.Metro.Shared/Controls/Dialogs/BaseMetroDialog.cs

```diff
@@ -88,25 +89,32 @@ static BaseMetroDialog()
         /// <param name="settings">The settings for the message dialog.</param>
         protected BaseMetroDialog(MetroWindow owningWindow, MetroDialogSettings settings)
         {
-            this.DialogSettings = settings ?? owningWindow.MetroDialogOptions;
-
-            this.OwningWindow = owningWindow;
-
-            this.Initialize();
+            this.Initialize(owningWindow, settings);
         }
 
         /// <summary>
         /// Initializes a new MahApps.Metro.Controls.BaseMetroDialog.
         /// </summary>
         protected BaseMetroDialog()
+            : this(null, new MetroDialogSettings())
         {
-            this.DialogSettings = new MetroDialogSettings();
+        }
 
-            this.Initialize();
+        /// <summary>
+        /// With this method it's possible to return your own settings in a custom dialog.
+        /// </summary>
+        /// <param name="settings"></param>
+        /// <returns></returns>
+        protected virtual MetroDialogSettings OnGetDialogSettings(MetroDialogSettings settings)
```

Comment:
I would call this method `ConfigureSettings`. E.g. in ASP.NET you have methods like `ConfigureServices`, which allow you to intercept and add additional services. Here it allows the user to intercept and change the settings object.